bip-0360.mediawiki

```diff
@@ -1,705 +1,401 @@
-because the MAX_SCRIPT_ELEMENT_SIZE=520. This is problematic because PQ signature algorithms often have signatures and 
-public keys in excess of 520 bytes. For instance:
-
-* ML-DSA public keys are 1,312 bytes and signatures are 2,420 bytes
-* SLH-DSA public keys are 32 bytes and signatures are 7,856 bytes
+The initial stack elements provide the same functionality as they do in a P2TR script-path spend. That is, they place elements on the stack to be evaluated by the redeem script.
 
-We will first look at our approach to the problem of PQ signatures and then give our solution for public keys larger than 520 bytes.
+The control block is a 1 + 32*d byte array, where the first byte is the control byte and the next 32*d bytes are the Merkle path to the leaf script, where d is the depth of the leaf script in the Merkle tree. The control byte is the same as the control byte in a P2TR control block, including the 7 bits which are used to specify the tapleaf version. The parity bit of the control byte is always 1, since P2TSH does not have a key-path spend. Unlike P2TR, we omit the public key from the control block as it is not needed in P2TSH. We maintain support for the optional annex in the witness (see Specification section below for more details).
 
-To keep P2QRH small and simple, we have opted not to raise the stack element size limit as part of P2QRH, but instead make this change when 
-adding PQ signatures. That said, we are not strongly opposed to putting this increase in P2QRH.
-
-We propose a stack element size limit of 8,000 bytes. We arrive at 8,000 by rounding up from the needed 7,856 bytes. 
+==Rationale==
 
-OP_DUP will duplicate any stack element. Thus, if we allowed OP_DUP to duplicate stack elements of size 8,000 bytes, it would be possible 
-to write a tapscript which will duplicate stack elements until it reaches the maximum number of elements on stack, i.e. 1000 elements. 
-An increase from 520 bytes to 8,000 bytes would increase the memory footprint from 520 KB to 8 MB.
+In this proposal, we adopt a “prepared not scared” approach to the possible advancement of quantum computing – and offer Bitcoin users an option for increased protection if they so choose.
 
-To prevent OP_DUP from creating an 8 MB stack by duplicating stack elements larger than 520 bytes we define OP_DUP to fail on stack 
-elements larger than 520 bytes. Note this change to OP_DUP is not consensus critical and does not require any sort of fork. This is
-because currently there is no way to get a stack element larger than 520 bytes onto the stack so triggering this rule is currently
-impossible and would only matter if the stack element size limit was raised.
+We understand that this subject is controversial and veiled in uncertainty; and hope to offer users flexible options - tailored to their level of concern - that are relatively unobtrusive to existing network architecture.
```

Comment:
I would cut this. The purpose of the BIP to propose a design and justify the design decisions. This reads more like us saying to the read "we hear you." I don't think a standard is the right place to communicate such things.

Reply:
Let's leave as is, Isabel will look at rewording

"Our motivation is user freedom and user choice..."

"Our motivation for the BIP is to enable users to protect themselves from the threats they are concerned"